src/Content/Text/BBCode.php

```diff
@@ -1252,647 +1252,635 @@ public static function cleanPictureLinks($text)
+				// sanitize href attributes (only allowlisted protocols URLs)
+				// default value for backward compatibility
+				$allowed_link_protocols = DI::config()->get('system', 'allowed_link_protocols', []);
 
-		array_walk($allowed_src_protocols, function(&$value) { $value = preg_quote($value, '#');});
+				// Always allowed protocol even if config isn't set or not including it
+				$allowed_link_protocols[] = '//';
+				$allowed_link_protocols[] = 'http://';
+				$allowed_link_protocols[] = 'https://';
+				$allowed_link_protocols[] = 'redir/';
 
-		$text = preg_replace('#<([^>]*?)(src)="(?!' . implode('|', $allowed_src_protocols) . ')(.*?)"(.*?)>#ism',
-					 '<$1$2=""$4 data-original-src="$3" class="invalid-src" title="' . DI::l10n()->t('Invalid source protocol') . '">', $text);
+				array_walk($allowed_link_protocols, function(&$value) { $value = preg_quote($value, '#');});
 
-		// sanitize href attributes (only allowlisted protocols URLs)
-		// default value for backward compatibility
-		$allowed_link_protocols = DI::config()->get('system', 'allowed_link_protocols', []);
+				$regex = '#<([^>]*?)(href)="(?!' . implode('|', $allowed_link_protocols) . ')(.*?)"(.*?)>#ism';
+				$text = preg_replace($regex, '<$1$2="javascript:void(0)"$4 data-original-href="$3" class="invalid-href" title="' . DI::l10n()->t('Invalid link protocol') . '">', $text);
 
-		// Always allowed protocol even if config isn't set or not including it
-		$allowed_link_protocols[] = '//';
-		$allowed_link_protocols[] = 'http://';
-		$allowed_link_protocols[] = 'https://';
-		$allowed_link_protocols[] = 'redir/';
+				// Shared content
+				$text = self::convertShare(
+					$text,
+					function (array $attributes, array $author_contact, $content, $is_quote_share) use ($simple_html) {
+						return self::convertShareCallback($attributes, $author_contact, $content, $is_quote_share, $simple_html);
+					}
+				);
 
-		array_walk($allowed_link_protocols, function(&$value) { $value = preg_quote($value, '#');});
+				$text = self::interpolateSavedImagesIntoItemBody($text, $saved_image);
 
-		$regex = '#<([^>]*?)(href)="(?!' . implode('|', $allowed_link_protocols) . ')(.*?)"(.*?)>#ism';
-		$text = preg_replace($regex, '<$1$2="javascript:void(0)"$4 data-original-href="$3" class="invalid-href" title="' . DI::l10n()->t('Invalid link protocol') . '">', $text);
+				return $text;
+			}); // Escaped noparse, nobb, pre
 
-		// Shared content
-		$text = self::convertShare(
-			$text,
-			function (array $attributes, array $author_contact, $content, $is_quote_share) use ($simple_html) {
-				return self::convertShareCallback($attributes, $author_contact, $content, $is_quote_share, $simple_html);
-			}
-		);
+			// Remove escaping tags
+			$text = preg_replace("/\[noparse\](.*?)\[\/noparse\]/ism", '\1', $text);
+			$text = preg_replace("/\[nobb\](.*?)\[\/nobb\]/ism", '\1', $text);
+			$text = preg_replace("/\[pre\](.*?)\[\/pre\]/ism", '\1', $text);
 
-		if ($saved_image) {
```

Comment:
Is the check for `$saved_image` removed?

Reply:
Yes, but the image interpolation is still here: https://github.com/friendica/friendica/pull/8729/files/7d800c81ecd616abb25c8543add1c9212809906c#diff-dbbffc32752b4d95e98eec53ad7d7f77

Reply:
But the check is removed :-) Shouldn't the function only be called when there is something to do?

Reply:
Not necessarily. The method calls foreach which doesn't do anything on empty arrays.